addons/mail/models/mail_link_preview.py

```diff
@@ -49,42 +46,39 @@ def _create_from_message_and_notify(self, message, request_url=None):
         for url in urls:
             if ignore_pattern and ignore_pattern.match(url):
                 continue
+            existing_link_preview = link_previews.search([("source_url", "=", url)])
+            if existing_link_preview and not url in link_previews_by_url:
+                self.env["mail.link.preview.message"].create({
+                    "message_id": message.id,
+                    "link_preview_id": existing_link_preview[0].id
+                })
+                link_previews += existing_link_preview
+                continue
             if url in link_previews_by_url:
                 preview = link_previews_by_url.pop(url)
-                if not preview.is_hidden:
-                    link_previews += preview
+                link_previews += preview
                 continue
             if preview := get_link_preview_from_url(url, requests_session):
-                preview['message_id'] = message.id
                 link_preview_values.append(preview)
             if len(link_preview_values) + len(link_previews) > 5:
                 break
         for unused_preview in link_previews_by_url.values():
-            unused_preview._unlink_and_notify()
+            self.env["mail.link.preview.message"].search([
```

Comment:
why not searching for every `mail.link.preview.message` at once?

Reply:
Globally agree, see above comment, try to be multi-enabled.